flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -169,18 +173,27 @@ public PackagedProgram(File jarFile, @Nullable String entryPointClassName, Strin
 	 *         may be a missing / wrong class or manifest files.
 	 */
 	public PackagedProgram(File jarFile, List<URL> classpaths, @Nullable String entryPointClassName, String... args) throws ProgramInvocationException {
-		if (jarFile == null) {
-			throw new IllegalArgumentException("The jar file must not be null.");
+		// Whether the job is a Python job and PythonRunner is the entry point.
+		if (entryPointClassName != null && entryPointClassName.equals(PythonDriver.class.getCanonicalName())) {
```

Comment:
Can be simplified to `isPython = entryPointClassName != null && entryPointClassName.equals(PythonDriver.class.getCanonicalName())`

Reply:
Yes.It looks better.